File: rustbridge-java/rustbridge-core/src/main/java/com/rustbridge/LogLevel.java
package com.rustbridge;

import org.jetbrains.annotations.NotNull;

/**
 * Log levels for plugin logging.
 */
public enum LogLevel {
    /**
     * Trace level - very detailed debugging information.
     */
    TRACE(0),

    /**
     * Debug level - debugging information.
     */
    DEBUG(1),

    /**
     * Info level - general information.
     */
    INFO(2),

    /**
     * Warn level - warning messages.
     */
    WARN(3),

    /**
     * Error level - error messages.
     */
    ERROR(4),

    /**
     * Off - disable logging.
     */
    OFF(5);

    private final int code;

    LogLevel(int code) {
        this.code = code;
    }

    /**
     * Get the level from a numeric code.
     *
     * @param code the level code
     * @return the corresponding level
     */
    public static @NotNull LogLevel fromCode(int code) {
        for (LogLevel level : values()) {
            if (level.code == code) {
                return level;
            }
        }
        return OFF;
    }

    /**
     * Get the numeric code for this level.
     *
     * @return the level code
     */
    public int getCode() {
        return code;
    }
}

File: rustbridge-csharp/RustBridge.Core/LogLevel.cs
namespace RustBridge;

/// <summary>
/// Log levels for plugin logging.
/// </summary>
public enum LogLevel
{
    /// <summary>
    /// Trace level - very detailed debugging information.
    /// </summary>
    Trace = 0,

    /// <summary>
    /// Debug level - debugging information.
    /// </summary>
    Debug = 1,

    /// <summary>
    /// Info level - general information.
    /// </summary>
    Info = 2,

    /// <summary>
    /// Warn level - warning messages.
    /// </summary>
    Warn = 3,

    /// <summary>
    /// Error level - error messages.
    /// </summary>
    Error = 4,

    /// <summary>
    /// Off - disable logging.
    /// </summary>
    Off = 5
}

/// <summary>
/// Extension methods for <see cref="LogLevel"/>.
/// </summary>
public static class LogLevelExtensions
{
    /// <summary>
    /// Get the level from a numeric code.
    /// </summary>
    /// <param name="code">The level code.</param>
    /// <returns>The corresponding level, or <see cref="LogLevel.Off"/> if invalid.</returns>
    public static LogLevel FromCode(int code)
    {
        if (Enum.IsDefined(typeof(LogLevel), code))
        {
            return (LogLevel)code;
        }
        return LogLevel.Off;
    }
}
